src/mlpack/methods/linear_svm/linear_svm.hpp

```diff
@@ -0,0 +1,250 @@
+/**
+ * @file linear_svm.hpp
+ * @author Ayush Chamoli
+ *
+ * An implementation of Linear SVM.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_LINEAR_SVM_LINEAR_SVM_HPP
+#define MLPACK_METHODS_LINEAR_SVM_LINEAR_SVM_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <ensmallen.hpp>
+
+#include "linear_svm_function.hpp"
+
+namespace mlpack {
+namespace svm {
+
+/**
+ * The LinearSVM class implements an L2-regularized support vector machine
+ * model, and supports training with multiple optimizers and classification.
+ * The class supports different observation types via the MatType template
+ * parameter; for instance, support vector classification can be performed
+ * on sparse datasets by specifying arma::sp_mat as the MatType parameter.
+ *
+ * Linear SVM can be used for general classification tasks which will work
+ * on multiclass classification. More technical details about
+ * the model can be found from the following:
+ *
+ * @code
+ * @inproceedings{weston1999support,
+ * title        = {Support vector machines for multi-class pattern
+ *                 recognition.},
+ * author       = {Weston, Jason and Watkins, Chris and others},
+ * booktitle    = {Esann},
```

Comment:
If you can fix the ESANN conference description here, it would be great; otherwise I can do it during merge.

Reply:
I'm not sure what needs to be changed here. In a previous review, you said ESANN is not a conference. Do I need to replace it with `The 7th European Symposium on Artificial Neural Networks`?
If any other change is needed here, I could make them but it is also fine if you could do these changes while merging.

Reply:
Yeah technically it would be "Proceedings of The 7th European Symposiun on Artifical Neural Networks". Anyway I'll handle it during merge.